### PR TITLE
use Error::source() and edition 2018 and bump to 2.0.0

### PR DESCRIPTION
**Pull request by haraldh**

### CODE REVIEW

**Comment by haraldh** [comment]
What do you think? Too much?

File: src/lib.rs [hunk 1]
```diff
@@ -383,8 +382,8 @@ macro_rules! quick_error {
                 self.0.description()
             }
             #[allow(deprecated)]
```
Comment (rossmacarthur) [inline src/lib.rs]: This can be removed.

File: src/lib.rs [hunk 2]
```diff
@@ -1022,16 +1020,14 @@ impl<T, E> ResultExt<T, E> for Result<T, E> {
     }
 }
 
-
-
 #[cfg(test)]
 #[allow(deprecated)]
```
Comment (rossmacarthur) [inline src/lib.rs]: This can be removed, and placed on the last test in this file only.

File: Cargo.toml [hunk 3]
```diff
@@ -4,10 +4,11 @@ description = """
...
 license = "MIT/Apache-2.0"
-version = "1.2.2"
+version = "2.0.0"
 keywords = ["macro", "error", "type", "enum"]
 authors = ["Paul Colomiets <paul@colomiets.name>", "Colin Kiegel <kiegel@gmx.de>"]
 homepage = "http://github.com/tailhook/quick-error"
 repository = "http://github.com/tailhook/quick-error"
 documentation = "http://docs.rs/quick-error"
 categories = ["rust-patterns"]
+edition = "2018"
```
Comment (rossmacarthur) [inline Cargo.toml]: A newline here would be nice.

**Review by rossmacarthur — COMMENTED** [review]
Hey, made some comments, I really hope @tailhook merges this soon.

In the meantime, I forked your version and published it as `quick-error2`, I hope that is okay.

File: Cargo.toml (hunk 3 above)
Comment (haraldh) [reply Cargo.toml]: done

**Comment by haraldh** [comment]
@rossmacarthur fixed your review comments

**Comment by tailhook** [comment]
Merged in b26770f. Sorry for delay

(I've messed up a bit when merging branch, and also broke a master branch of your repo sorry for that)

**Pull request closed without merging**